pkg/detectors/github/v1/github_old.go

```diff
@@ -72,63 +77,20 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		s1 := detectors.Result{
 			DetectorType: detectorspb.DetectorType_Github,
 			Raw:          []byte(token),
-		}
-		s1.ExtraData = map[string]string{
-			"rotation_guide": "https://howtorotate.com/docs/tutorials/github/",
-			"version":        fmt.Sprintf("%d", s.Version()),
+			ExtraData: map[string]string{
+				"rotation_guide": "https://howtorotate.com/docs/tutorials/github/",
+				"version":        fmt.Sprintf("%d", s.Version()),
+			},
 		}
 
 		if verify {
 			client := common.SaneHttpClient()
-			// https://developer.github.com/v3/users/#get-the-authenticated-user
-			for _, url := range s.Endpoints(s.DefaultEndpoint()) {
-				req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s/user", url), nil)
-				if err != nil {
-					continue
-				}
-				req.Header.Set("Content-Type", "application/json; charset=utf-8")
-				req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
-				res, err := client.Do(req)
-				if err == nil {
-					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						var userResponse userRes
-						err = json.NewDecoder(res.Body).Decode(&userResponse)
-						res.Body.Close()
-						if err == nil {
-							s1.Verified = true
-
-							if err == nil {
-								s1.Verified = true
-								s1.ExtraData["username"] = userResponse.Login
-								s1.ExtraData["url"] = userResponse.UserURL
-								s1.ExtraData["account_type"] = userResponse.Type
-								if userResponse.SiteAdmin {
-									s1.ExtraData["site_admin"] = "true"
-								}
-								if userResponse.Name != "" {
-									s1.ExtraData["name"] = userResponse.Name
-								}
-								if userResponse.Company != "" {
-									s1.ExtraData["company"] = userResponse.Company
-								}
-								if userResponse.LdapDN != "" {
-									s1.ExtraData["ldap_dn"] = userResponse.LdapDN
-								}
-
-								// GitHub does not seem to consistently return this header.
-								scopes := res.Header.Get("X-OAuth-Scopes")
-								if scopes != "" {
-									s1.ExtraData["scopes"] = scopes
-								}
-								expiry := res.Header.Get("github-authentication-token-expiration")
-								if expiry != "" {
-									s1.ExtraData["expires_at"] = expiry
-								}
-							}
-						}
-					}
-				}
-			}
+
+			isVerified, userResponse, headers, err := s.VerifyGithub(ctx, client, token)
```

Comment:
question: Did you consider returning ptrs for `userResponse` and `headers`? If we return pointers you can more easily handle the nil check so you don't have to do the comparison to empty structs in both `SetUserResponse` and `SetHeaderInfo`.

You could do something like:
```
if userResponse != nil {
  SetuserResponse(*userResponse, &s1)
}
if headers != nil {
  SetHeaderInfo(*headers, &s1)
}
```